list/management.cattle.io.setting.vue

```diff
@@ -0,0 +1,157 @@
+<script>
+import { mapGetters } from 'vuex';
+import { MANAGEMENT } from '@/config/types';
+import { ALLOWED_SETTINGS } from '@/config/settings';
+import Banner from '@/components/Banner';
+import { DEV } from '@/store/prefs';
+
+export default {
+  components: { Banner },
+
+  async fetch() {
+    const isDev = this.$store.getters['prefs/get'](DEV);
+    const rows = await this.$store.dispatch(`management/findAll`, { type: MANAGEMENT.SETTING });
+    const t = this.$store.getters['i18n/t'];
+    // Map settings from array to object keyed by id
+    const settingsMap = rows.reduce((res, s) => {
+      res[s.id] = s;
+
+      return res;
+    }, {});
+
+    const settings = [];
+
+    // Combine the allowed settings with the data from the API
+    Object.keys(ALLOWED_SETTINGS).forEach((setting) => {
+      const s = {
+        ...ALLOWED_SETTINGS[setting],
+        id:          setting,
+        description: t(`advancedSettings.descriptions.${ setting }`),
+        data:        settingsMap[setting],
+        customized:  !!settingsMap[setting].value && !!settingsMap[setting].default
+      };
+
+      s.hide = s.canHide = (s.kind === 'json' || s.kind === 'multiline');
+      if (s.kind === 'json') {
+        s.json = JSON.stringify(JSON.parse(s.data.value || s.data.default), null, 2);
+      } else if (s.kind === 'enum') {
+        const v = s.data.value || s.data.default;
+
+        s.enum = `advancedSettings.enum.${ setting }.${ v }`;
+      }
+      // There are only 2 actions that can be enabled - Edit Setting or View in API
+      // If neither is available for this setting then we hide the action menu button
+      s.hasActions = !s.readOnly || isDev;
+      settings.push(s);
+    });
+    this.settings = settings;
+  },
+
+  data() {
+    return { settings: null };
+  },
+
+  computed: {
+    ...mapGetters({ t: 'i18n/t' }),
+
+    rows() {
```

Comment:
Just for personal knowledge, would wrapping content with the Loading/fetch pending component mean `this.settings` could be used directly?

Reply:
Seems to work that way